File: component/hub/hubproxy.cs
using System;
using System.Collections.Generic;
using System.Collections;

namespace hub
{
    public class hubproxy
    {
        public hubproxy(String hub_name, juggle.Ichannel ch)
        {
            name = hub_name;
            _caller = new caller.hub_call_hub(ch);
        }

        public void reg_hub_sucess()
        {
            _caller.reg_hub_sucess();
        }

        public void caller_hub(string module_name, string func_name, ArrayList argvs)
        {
            _caller.hub_call_hub_mothed(module_name, func_name, argvs);
        }

        public string name;
        private caller.hub_call_hub _caller;
    }
}

File: rpc/test_python/hubproxy.py
import msgpack
import abelkhan
import hub

class hubproxy(object):
    def __init__(self, hub_name:str, hub_type:str, _ch:abelkhan.Ichannel, _modulemng:abelkhan.modulemng) -> None:
        self.name = hub_name
        self.type = hub_type
        self.ch = _ch
        self.hub_call_hub_caller = hub.hub_call_hub_caller(self.ch, _modulemng)
    
    def caller_hub(self, func_name:str, argvs:list):
        _event = [func_name, argvs]
        self.hub_call_hub_caller.hub_call_hub_mothed(msgpack.dumps(_event))
